### PR TITLE
fix node role <none>

### PR DESCRIPTION
**Pull request by snowplayfire**
**What type of PR is this?**
> /kind api-change
> /kind bug

**What this PR does / why we need it**:
When upgrade to v1.17.3 from v1.13.2,  node roles show as `<none>`
```
root@xxx-MASTER01 :~  # kubectl get node node1
NAME     STATUS   ROLES    AGE    VERSION
node1    Ready    <none>   3h4m   v1.17.3-trip
```

**Which issue(s) this PR fixes**:
In former k8s version,  the node-label of `node-role.kubernetes.io` is allowed.   And kubectl displays node roles using the the label.

In v1.17.3,  the label prefix `node-role.kubernetes.io` is not allowed, suggesting `node.kubernetes.io` instead. But kubectl displays node roles still based on the former prefix `node-role.kubernetes.io`.  So kubectl displays node roles as `<none>`.

As https://github.com/kubernetes/enhancements/blob/master/keps/sig-auth/0000-20170814-bounding-self-labeling-kubelets.md#proposal, I suggest that `node-role.kubernetes.io` can be changed to `role.node.kubernetes.io`, `kubernetes.io/role` can be changed to `kubelet.kubernetes.io/role`.

Then we can get the following after label the node `role.node.kubernetes.io/master=true` or `kubelet.kubernetes.io/role=master`
```
root@xxx-MASTER01 :~  # kubectl get node node1
NAME     STATUS   ROLES    AGE    VERSION
node1    Ready    master   3h4m   v1.17.3-trip
```

Fixes #

**Special notes for your reviewer**:
```
Environment:
kubernetes v1.17.3
```

### CODE REVIEW

File: pkg/printers/internalversion/printers.go [hunk 1]
```diff
@@ -72,10 +72,12 @@ const (
 
 	// labelNodeRolePrefix is a label prefix for node roles
 	// It's copied over to here until it's merged in core: https://github.com/kubernetes/kubernetes/pull/39112
-	labelNodeRolePrefix = "node-role.kubernetes.io/"
+	labelNodeRolePrefix           = "role.node.kubernetes.io/"
```
Comment (soltysh) [inline pkg/printers/internalversion/printers.go:75]: Where this is coming from? I can't seem to find that label anywhere in the code?

**Review by soltysh — CHANGES_REQUESTED** [review]
/assign

File: pkg/printers/internalversion/printers.go (hunk 1 above)
Comment (willthames) [reply pkg/printers/internalversion/printers.go:75]: From the description of this PR:
> I suggest that `node-role.kubernetes.io` can be changed to `role.node.kubernetes.io`, `kubernetes.io/role` can be changed to `kubelet.kubernetes.io/role`

Also in the description:
> In v1.17.3, the label prefix `node-role.kubernetes.io` is not allowed, suggesting `node.kubernetes.io` instead.

That one definitely needs to be prefixed as `node.kubernetes.io` is used for other purposes (e.g. `node.kubernetes.io/instance-type`) - as all prefixes are allowed by https://github.com/kubernetes/enhancements/blob/master/keps/sig-auth/0000-20170814-bounding-self-labeling-kubelets.md#proposal, `role.node.kubernetes.io` seems a great choice)

File: pkg/printers/internalversion/printers.go [hunk 2]
```diff
@@ -1538,6 +1540,9 @@ func getNodeInternalIP(node *api.Node) string {
 
 // findNodeRoles returns the roles of a given node.
 // The roles are determined by looking for:
+// * a role.node.kubernetes.io/<role>="" label
+// * a kubelet.kubernetes.io/role="<role>" label
```
Comment (willthames) [inline pkg/printers/internalversion/printers.go:1544]: I wonder if `role.kubelet.kubernetes.io/<role>=""` is more consistent with the node approach?

Comment (snowplayfire) [reply pkg/printers/internalversion/printers.go]: Agree, I have updated the PR.    Please review it again, Thanks!

**Pull request closed without merging**